lib/gateway/index.js

```diff
@@ -20,12 +20,19 @@ module.exports = function ({ plugins, config } = {}) {
   const { httpServer, httpsServer } = bootstrap({ plugins, config });
 
   if (config.gatewayConfig.http && httpServer) {
+    // config.gatewayConfig.http.hostname = config.gatewayConfig.http.hostname || 'localhost';
     appPromises.push(
       new Promise(resolve => {
-        const runningApp = httpServer.listen(config.gatewayConfig.http.port, () => {
-          const { address, port } = runningApp.address();
+        const runningApp = httpServer.listen(config.gatewayConfig.http.port, config.gatewayConfig.http.hostname, () => {
+          // This can either be an object or a String depending on whether you specify a host or not
+          // https://nodejs.org/api/net.html#net_server_address
+          if (config.gatewayConfig.http.hostname) {
```

Comment:
Instead of checking our configuration values, I'd rather check the values returned by the `address()` function to decide what to print or not.

```javascript
            const addressInfo = runningApp.address();
            if (typeof addressInfo === 'string') {
              log.info(`gateway http server listening on ${addressInfo}`);
            } else {
              log.info(`gateway http server listening on ${addressInfo.address}:${addressInfo.port}`);
            }
```